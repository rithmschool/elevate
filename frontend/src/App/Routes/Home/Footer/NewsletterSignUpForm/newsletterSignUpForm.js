import React from "react";
import { Form, FormGroup, Label, Input } from "reactstrap";

import ElevateApi from "../../../../../elevateApi";

class newsletterSignUpForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      signedUp: false,
      emailExists: false
    };
  }

  handleChange = evt => {
    this.setState({ [evt.target.name]: evt.target.value });
  };

  handleSubmit = async evt => {
    evt.preventDefault();
    const newsletter = await ElevateApi.getNewsletterSignUp(this.state);
    if (!newsletter.email) {
      await ElevateApi.postNewsletterSignUp(this.state);
      this.setState({
        email: "",
        signedUp: true,
        emailExists: false
      });
    } else {
      this.setState({
        email: "",
        signedUp: false,
        emailExists: true
      });
    }
  };

  render() {
    const newsletterSignUp = (
      <div>
        <h6>Stay up to date with Brella</h6>
        <Form onSubmit={this.handleSubmit}>
          <FormGroup>
            <Label for="email" hidden>
              newsletter email
            </Label>
            <Input
              type="text"
              id="email"
              name="email"
              placeholder="Email Address"
              value={this.state.email}
              onChange={this.handleChange}
            />
          </FormGroup>
        </Form>
      </div>
    );

    return (
      <div>
        {this.state.signedUp ? (
          <h6>Thanks for signing up!</h6>
        ) : (
          newsletterSignUp
        )}
        {this.state.emailExists && (
          <p className="text-danger">This email is already signed up</p>
        )}
      </div>
    );
  }
}

export default newsletterSignUpForm;
