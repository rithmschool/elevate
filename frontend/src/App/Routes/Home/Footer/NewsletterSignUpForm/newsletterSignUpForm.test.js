import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import NewsletterSignUpForm from "./newsletterSignUpForm";

describe("NewsletterSignUpForm", function() {
  it("renders", () => {
    shallow(<NewsletterSignUpForm />);
  });

  it("matches snapshot", () => {
    let wrapper = mount(<NewsletterSignUpForm />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it("calls onSubmit prop function when form is submitted", () => {
    const submitFn = jest.fn();
    const wrapper = mount(<NewsletterSignUpForm onSubmit={submitFn} />);
    const form = wrapper.find("form");
    console.log(form.debug());
    form.simulate("change", { email: { value: "test@gmail.com" } });
    form.simulate("submit");
    expect(submitFn).toHaveBeenCalledTimes(1);
  });
});
