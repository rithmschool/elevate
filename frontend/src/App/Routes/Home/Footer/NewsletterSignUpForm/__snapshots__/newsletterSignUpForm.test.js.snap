// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsletterSignUpForm matches snapshot 1`] = `
<newsletterSignUpForm>
  <div>
    <div>
      <h6>
        Stay up to date with Brella
      </h6>
      <Form
        inline={false}
        onSubmit={[Function]}
      >
        <form
          className=""
          onSubmit={[Function]}
        >
          <FormGroup>
            <div
              className="form-group"
            >
              <FormLabel
                column={false}
                hidden={true}
                htmlFor="email"
                srOnly={false}
              >
                <label
                  className="form-label"
                  hidden={true}
                  htmlFor="email"
                >
                  newsletter email
                </label>
              </FormLabel>
              <FormControl
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="text"
                value=""
              >
                <input
                  className="form-control"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email Address"
                  type="text"
                  value=""
                />
              </FormControl>
            </div>
          </FormGroup>
        </form>
      </Form>
    </div>
  </div>
</newsletterSignUpForm>
`;
